src/compiler/checker.ts

```diff
@@ -22962,9 +22962,16 @@ namespace ts {
                                 case SyntaxKind.AsteriskAsteriskToken: return left ** right;
                             }
                         }
+                        if (typeof left === "string" && typeof right === "string") {
+                            switch ((<BinaryExpression>expr).operatorToken.kind) {
+                                case SyntaxKind.PlusToken: return left + right;
+                            }
+                        }
                         break;
                     case SyntaxKind.StringLiteral:
                         return (<StringLiteral>expr).text;
+                    case SyntaxKind.NoSubstitutionTemplateLiteral:

```

Comment:
make sure `getEnumKind` handles `NoSubstitutionTemplateLiteral` correctly, and add a test for an literal enum with only template literals. I would recommend doing this in a separate PR instead, since this is not really the change originally intended.